cpp/src/Ice/Selector.cpp

```diff
@@ -639,7 +639,10 @@ Selector::select(int timeout)
     }
     else if (timeout > 0)
     {
+        // kpoll and select use seconds, epoll and poll use milliseconds
+#    if !defined(ICE_USE_KQUEUE) && !defined(ICE_USE_SELECT)
```

Comment:
When timespec is used we actually need seconds, no milliseconds. AFAICT this also affects 3.7.